### DISCUSSION TITLE
✅ Track UTM parameters

### DISCUSSION THREAD

**Discussion by dannyvankooten** [ibericode/koko-analytics#209] [Ideas]
### Update (May 2026): this is now natively supported in [Koko Analytics Pro](https://www.kokoanalytics.com/features/#pro).
Read more: [Tracking UTM parameters in Koko Analytics Pro](https://www.kokoanalytics.com/docs/tracking/tracking-utm-parameters/)

---

> > Could it be possible to integrate UTM parameters in Koko Analytics statistics?
> 
> Right now I cannot figure out what social media or specific campaign is working for my (non-commercial) blog. At least in my case, referrer has very few data and is not that useful.
> 
> Imho it will be good both for non-commercial and commercial websites, as this way it would be possible to understand where resources (time, attention, and often money) are used effectively.
> 
> Pros:
> 
>     It will be possible to determine how effective social and ads campaign are
>     UTM parameters are a common standard
>     Non-invasive tracking
> 
> Cons:
> 
>     Koko will became more complicated (possible solution: only include utm_source and utm_campaign)
>     Performance slowdown (possible solution: checkbox to disable this function)
>  @davleone in #118

**Comment by michacassola** [comment]
Any news on this? Thanks! 👍

**Comment by dannyvankooten** [comment]
Koko Analytics Pro version 2.3.3 and up now contains support for UTM tracking. Once enabled, the referrers component will show campaigns, mediums and sources.

<img width="869" height="250" alt="image" src="https://github.com/user-attachments/assets/b8cab67f-7d27-4a91-aaff-8afd877a5b91" />

https://pro.kokoanalytics.com/plugin/changelog
